### ISSUE TITLE
Add custom mime_types fails - config loadView passing all mime_type values to MimeTypeListener

### ISSUE THREAD

**Issue by fiburonsk** [FriendsOfSymfony/FOSRestBundle#1340]
I am trying to add custom mime_types for json and xml in my view configuration per the documentation [http://symfony.com/doc/current/bundles/FOSRestBundle/3-listener-support.html#mime-type-listener](http://symfony.com/doc/current/bundles/FOSRestBundle/3-listener-support.html#mime-type-listener) something like:

    view:
        formats:
            json: true
            xml: true
        mime_types:
            json: ['application/vnd.test+json']
            xml: ['application/vnd.test+xml']
        ...

I keep receiving the following error:

> Format 'enabled' not supported, handler must be implemented
> 
> 415 Unsupported Media Type - UnsupportedMediaTypeHttpException
> in FOSRestExtension::loadView line 264

after sending a GET request with header `Accept: application/vnd.test+json`.  

Everything from view.mime_types is being passed into the MimeTypeListener constructor which in my case looks like

```
FOS\RestBundle\EventListener\MimeTypeListener:
array:3 [▼
  "enabled" => true
  "formats" => array:2 [▼
    "json" => array:1 [▶]
    "xml" => array:1 [▶]
  ]
  "service" => null
]
```

I suspect the following as indicated in the exception:

    $container->getDefinition('fos_rest.mime_type_listener')->replaceArgument(0, $config['view']['mime_types']);

Should be:

    $container->getDefinition('fos_rest.mime_type_listener')->replaceArgument(0, $config['view']['mime_types']['formats']);

Making the above change fixes the problem for me, but overwrites any existing mime_types generated from Request::initializeFormats().  If overwriting those defaults is desired then I'd say this fixes it.  If keeping the defaults is preferred then the MimeTypeListener will need a way to merge them instead of simply calling `$request->setFormat($format, $mimeType);` on line 57 of MimeTypeListener.

thanks!

**Comment by Ener-Getick** [comment]
Good catch :+1: Which version of FOSRestBundle are you using?

**Comment by Ener-Getick** [comment]
Ok in fact, that's fixed in https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1317, we just need to merge it

**Comment by fiburonsk** [comment]
oh ok.  I was using master for the version 2 stuff in my project.
thanks!

**Comment by Ener-Getick** [comment]
I'll work on a patch to have less BC breaks between FOSRestBundle 1.8 and 2.0 on the mime types ;-)

**Comment by Ener-Getick** [comment]
Fixed in #1317
